packages/effect/src/internal/core.ts

```diff
@@ -1136,47 +1136,54 @@ export const sync = <A>(evaluate: LazyArg<A>): Effect.Effect<never, never, A> =>
 /* @internal */
 export const tap = dual<
   {
-    <A, X>(
-      f: (a: NoInfer<A>) => X
+    <A, R1, E1>(
+      f: (a: NoInfer<A>) => Effect.Effect<R1, E1, unknown>
```

Comment:
Instead of `unknown`, using a type hole is safer (you can just add an unused generic like `X` or `_`)

Reply:
I'm curious about this. How is it safer? Any examples?

Reply:
It improves inference, otherwise TS may mistakenly consider it as a candidate for inference in generic contexts